array_methods_lab.js

```diff
@@ -0,0 +1,365 @@
+// Input: [1,2,3,4,5]
+// Output: 15
+
+console.log('========answer#5========')
+
+let arr4 = [1,2,3,4,5]
+let sum = arr4. reduce ((total, currentElem) => {
+  return total + currentElem;
+})
+console.log(sum)
+// ```
+//
+// 6. Write a function that finds largest number in an array.
+//
+// ```
+// Input: [1,2,6,3,4,5]
+// Output: [6]
+
+console.log('=======answer#6=======')
+
+let arr5 = [1,2,6,3,4,5]
+let largestNum = arr5 .reduce ((acc, currElem) => {
+    if (acc > currElem) {
+    return acc
+  } else {
+    return currElem
+  }
+})
+console.log(largestNum)
+// ```
+//
+// 7. Write a function `countZeroes`, which takes an array of numbers as its
+// argument and returns the amount of zeroes that occur in it.
+//
+// ```
+// Input: [1,0,0,2,0,3,0,0,0]
+// Output: 6
+
+console.log('=======answer#7=======')
+
+let arr6 = [1,0,0,2,0,3,0,0,0]
+
+let countZeroes = arr6.reduce((acc, currentElem) => (currentElem === 0) ?
+++ acc : acc, 0)
+
+console.log(countZeroes)
+
+
+
+// ```
+//
+// 8. Write a function that doubles every element in an array.
+//
+// ```
+// Input: [1,2,3,4]
+// Output: [2,4,6,8]
+
+console.log('========answer#8=========')
+
+let arr7 = [1,2,3,4,5]
+let doubles = []
+
+arr7.forEach((currElem) => {
+  doubles.push(currElem * 2)
+})
+console.log(doubles)
+// ```
+//
+// 9. Write a function that finds the sum of every element in an array doubled
+//
+// ```
+// Input: [1,2,3,4]
+// Output: 20
+
+console.log('=======answer#9=========')
+
+let arr8 = [1,2,3,4,5]
+
+let sumOfElem = arr8.reduce((total, currElem) => {
+  return total + (currElem * 2)
+}, 0)
+console.log(sumOfElem)
+// ```
+//
+// 10. Write a function `elementDivisibleBy` with parameters `divisor` and `arr`.
+// Return a new array of every element of arr that can be evenly divided by divisor.
+//
+// ```
+// Input: 5, [4,5,6,7,8,9,10,11,12,13,14,15]
+// Output: [5,10,15]
+
+console.log('=======answer#10======')
+
+let arr9 =  [4,5,6,7,8,9,10,11,12,13,14,15]
+let elementDivisibleBy = (divisor, arr) =>
+arr9.filter(elem => elem % divisor === 0)
+
+
+console.log(elementDivisibleBy(5, arr9))
+// ```
+//
+// 11. Write a `numberTimesIdx` that uses `map` and multiples each number in the
+// array by its index.
+//
+// ```
+// Input: [6,7,8,9]
+// Output: [0,7,16,27]
+
+console.log('=======answer#11=======')
+
+let arr10 = [6,7,8,9]
+let numberTimesIdx = arr10.map((currNum, index) => {
+  return currNum * index
+})
+console.log(numberTimesIdx)
+
+
+// ```
+//
+// 12. Write a function that returns whether or not every value in an array is a
+// positive even number that doesn't end in 0.
+//
+// ```
+// Input: [2,4,6,8,12]
+// Output: true
+//
+// Input: [2,4,6,8,10]
+// Output: false
+//
+console.log('======answer#12========')
+
+let arr11 = [2,4,6,8,12]
+let positiveEvenNum = arr11.every((currNum, index) => {
+  if (currNum === 'positive' || currNum % 2 === 0 || currNum % 10 !== 0 ) {
+    return true
+  } else {
+    return false
+  }
+
+})
+console.log(positiveEvenNum)
+
```

Comment:
Right idea .. So I've actually never heard of the keyword positive. But currNum > 0 would be a great substitute.